src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp

```
@@ -106,8 +117,15 @@ void FleetApiClient::sendCommand(const std::string &commandJson) {
 }
 
 
-void FleetApiClient::sendStatus(const std::string &statusJson) {
-	payloadPtr_->setMessageType(settings::Constants::STATUS_MESSAGE_TYPE);
+void FleetApiClient::sendStatus(const std::string &statusJson, StatusType statusType) {
+	switch(statusType) {
+		case StatusType::STATUS:
+			payloadPtr_->setMessageType(settings::Constants::STATUS_MESSAGE_TYPE);
+			break;
+		case StatusType::STATUS_ERROR:
+			payloadPtr_->setMessageType(settings::Constants::STATUS_ERROR_MESSAGE_TYPE);
+			break;
```

please add default and and throw an error "invalid_argument".

---

default with exception throw added